set-mongo/src/main/java/eu/europeana/set/mongo/service/PersistentUserSetServiceImpl.java

```diff
@@ -89,6 +92,26 @@ public PersistentUserSet getByIdentifier(String identifier) {
 		return getUserSetDao().findOne(query);
 	}
 	
+	/* (non-Javadoc)
+	 * @see eu.europeana.set.mongo.service.PersistentUserSetService#isTypeAndCreatorExisting(java.lang.String, eu.europeana.set.definitions.model.agent.Agent)
+	 */
+	public boolean isTypeAndCreatorExisting(String type, Agent creator) {
+	    Query<PersistentUserSet> query = getUserSetDao().createQuery();
+	    query.filter(PersistentUserSet.FIELD_TYPE, type);
+
+	    QueryResults<PersistentUserSet> results = getUserSetDao().find(query);
+	    List<PersistentUserSet> userSetList = results.asList();
+	    
+	    boolean res = false;
+	    for (UserSet userSet : userSetList) {
+		if (userSet.getCreator().getName().equals(creator.getName())) {
+		    res = true;
+		    break;
+		}
+	    }
+	    return res;
```

Comment:
the method must return the UserSetFolder

Reply:
we just check if given type and creator already exists

Reply:
the method must return the Object as this method may be used for multiple purposes.

Reply:
Refactored so that method returns object